cmd/podman/main.go

```diff
@@ -36,6 +37,16 @@ func main() {
 		return
 	}
 
+	if filepath.Base(os.Args[0]) == registry.PodmanSh ||
+		(len(os.Args[0]) > 0 && filepath.Base(os.Args[0][1:]) == registry.PodmanSh) {
+		shell := strings.TrimPrefix(os.Args[0], "-")
+		args := []string{shell, "exec", "-t", "-i", "--wait", "10", registry.PodmanSh, "/bin/sh"}
```

Comment:
```
--- a/cmd/podman/main.go
+++ b/cmd/podman/main.go
@@ -40,13 +40,23 @@ func main() {
        if filepath.Base(os.Args[0]) == registry.PodmanSh ||
                (len(os.Args[0]) > 0 && filepath.Base(os.Args[0][1:]) == registry.PodmanSh) {
                shell := strings.TrimPrefix(os.Args[0], "-")
-               args := []string{shell, "exec", "-t", "-i", "--wait", "10", registry.PodmanSh, "/bin/sh"}
+               args := []string{shell, "exec", "-i", "--wait", "10"}
+               if len(os.Args) == 1 {
+                       args = append(args, "-t")
+               }
+               args = append(args, registry.PodmanSh, "/bin/sh")
                if len(os.Args) > 1 {
                        args = append(args, os.Args[1:]...)
                }
                os.Args = args
```

For scp/sftp is important to run without attached tty. This is similar approach that sshd uses - tty is only attached when an interactive shell is run (or when it's enforced on client side)

e.g. command `scp file podmanuser@localhost:`  runs `ssh podmanuser@localhost sftp-server` and therefore the shell/podman needs to be run without tty attached.

The patch above tries to address this by checking the number of arguments:

- `ssh podmanuser@localhost`  means os.Args == 1
- `ssh podmanuser@localhost <command>` means that os.Args > 1

It would be probably better to detect whether tty is already attached or not and based on this decide whether run `podman exec` with or without `-t`

Reply:
yes I guess you want to actually check for a tty then not just a command,
this can be done with `term.IsTerminal(0) || term.IsTerminal(1) || term.IsTerminal(2)`

Reply:
Good point we should check if there is a terminal, and then remove the -t.  Should -i be removed if we don't have  term.IsTerminal(0)?

Reply:
No -i should always be set, in this shell context it will always be required. And passing -i does not harm even when it is not needed.

Reply:
PTAL at latest version of PR.

`scp file user@host:` works with @bachradsusi's fedora-selinux image but with `ubi9-minimal` it just closes connection without scp-ing. Petr did you make any changes to your image?

Reply:
`scp` is a wrapper around `ssh <host> /usr/libexec/openssh/sftp-server` therefore a target system/container needs to have `openssh-server` installed. On a standard host `sshd` and `sftp-server`  are installed at the same time, but in this case `sshd` runs in a different root than `sftp-server`.

In my image I use:
```
RUN dnf -y install selinux-policy-targeted openssh-clients openssh-server procps-ng podman && dnf clean all
```